Performance improvement options (#2230)
cloudflare/cloudflare-docs: products/pages/src/content/platform/performace-improvements.md

---
order: 10
pcx-content-type: concept
---

# Performance Improvements

## Polish 

> Polish improves image loading time by optimizing image hosted on your “domain.pages.dev”. It reduces image file size by removing metadata(date and time, camera manufacturer, model, etc.) by compressing images where possible,  smaller images mean faster load time for images and web apps. 

Reviewer: ```suggestion
Polish improves image loading time by optimizing image hosted on your “domain.pages.dev”. It reduces image file size by removing metadata (date and time, camera manufacturer, model, etc.) by compressing images where possible, smaller images mean faster load time for images and web apps.
```
Spacing

By default, the Polish option is turned off for projects, it does come with two other options that have different performance effects on images, namely Lossy and lossless.

### Lossless 

> Lossless image option reduces the size of image files without impacting the visual quality. In this option, metadata is removed from PNG, GIF and JPEG but the lossless compression is done on GIF and PNG

Reviewer: ```suggestion
Lossless image option reduces the size of image files without impacting the visual quality. In this option, metadata is removed from PNG, GIF and JPEG but the lossless compression is done on GIF and PNG.
```
Consistency

### Lossy 

> This polish option, reduces the file size of JPEG images using lossy compression which may reduce image quality 

Reviewer: ```suggestion
This polish option, reduces the file size of JPEG images using lossy compression which may reduce image quality.
```
Consistency

---
order: 10
pcx-content-type: concept
---

# Performance Improvements

Cloudflare Pages now includes performance improvements to serve pages of your projects by removing the overhead on media assets and offering compressions. 
 

> **Note: ** If you have a custom domain tied to your Pages project, configuring your performance settings in the Speed tab will override the performance settings configured in Pages. 

Reviewer 2: ```suggestion
<Aside type="note">

If you have a custom domain tied to your Pages project, configuring your performance settings in **Speed** on the Cloudflare dashbaord will override the performance settings configured in Pages. 

</Aside>
```

> ## Brotli compression.

Reviewer 2: ```suggestion
## Brotli compression
```

> Cloudflare applies Brotli compression to help speed up page load times for your visitors.  Brotli compression is selected as the preferred content encoding method if multiple compression methods are supported by the client. If the client does not indicate that Brotli is supported, the gzip compression will be applied. 

Reviewer 2: ```suggestion
Cloudflare applies [Brotli](https://www.brotli.pro/) compression to help speed up page load times for your visitors. Brotli compression is selected as the preferred content-encoding method if multiple compression methods are supported by the client. If the client does not indicate that Brotli is supported, the gzip compression will be applied. 
```

> This option is automatically turned on for all Pages sites and cannot be turned off through the Pages dashboard.

Reviewer 2: ```suggestion
The Brotli compression option is automatically turned on for all Pages sites and cannot be turned off in the **Pages** dashboard.
```

> However, if your project is tied to a custom domain, you may adjust your Brotli settings in the Speed tab. 

Reviewer 2: ```suggestion
However, if your project is tied to a custom domain, you may adjust your Brotli settings in **Speed** > **Optimization** > moving the **Brotli** toggle to **Off*.
```

> You can learn more about [how cloudflare handles compression](https://support.cloudflare.com/hc/en-us/articles/200168396-What-will-Cloudflare-compress-) in this support document.

Reviewer 2: ```suggestion
You can learn more about [how Cloudflare handles compression by reviewing this support article](https://support.cloudflare.com/hc/en-us/articles/200168396-What-will-Cloudflare-compress-).
```

## Polish 

> Polish improves image loading time by optimizing images hosted on your [domain].pages.dev. It reduces image file size by removing metadata (date and time, camera manufacturer, model, etc.) by compressing images where possible; smaller images mean faster load time for images and web apps. 

Reviewer 2: ```suggestion
Polish improves image loading time by optimizing images hosted on your `*.pages.dev`site. It reduces image file size by removing metadata (date and time, camera manufacturer, model, etc.) by compressing images where possible. Smaller images result in faster load time for images and web applications. 
```

> There are two options for configuration with different performance effects on images -- Lossless and Lossy.

Reviewer 2: ```suggestion
There are two options for configuration with different performance effects on images -- _Lossless_ and _Lossy_. You can choose between these two options by going to **Speed** > **Optimization** > **Polish**.
```

### Lossless 

Lossless image option reduces the size of image files without impacting the visual quality. In this option, metadata is removed from PNG, GIF and JPEG but the lossless compression is done on GIF and PNG.

### Lossy 

> This polish option, reduces the file size of JPEG images using lossy compression which may reduce image quality.

Reviewer 2: ```suggestion
This Polish option reduces the file size of JPEG images using lossy compression which may reduce image quality.
```

> > Making the decision on what form of optimization to use is mostly based on use case. There are a lot of resources to show [the differnce between lossy and lossless image compression](https://imagify.io/blog/lossless-vs-lossy-image-compression/).

Reviewer 2: ```suggestion
> Making the decision on what form of optimization to use is mostly based on your individual use case. You can review the variety of resources to learn more about [the difference between Lossy and Lossless image compression on Imagify's blog](https://imagify.io/blog/lossless-vs-lossy-image-compression/).
```

Reviewer 2: Is there a reason this has a > ?

Author: I was trying to highlight this section, I think the Aside component should be better for this.

### Serving WebP images 

> Whether you are choosing Lossy or Lossless, you also have the option of enabling WebP for your images. If the client requesting the image supports the WebP image codec, Cloudflare will serve a WebP version of the image if offering a performance advantage over the existing format. 

Reviewer 2: ```suggestion
Whether you choose _Lossy_ or _Lossless_, you also have the option of enabling WebP for your images. If the client requesting the image supports the WebP image codec, Cloudflare will serve a WebP version of the image if offering a performance advantage over the existing format. 
```

Reviewer 2: There are two 'if' clauses here which is w bit confusing. Can we reword this for clarity?

<Aside heading="Status Code">

> **Note: ** Large Jpeg images are converted to progressive images. Visitors will see increasingly detailed images as file is downloaded. 

Reviewer 2: ```suggestion
Large JPEG images are converted to progressive images with WebP enabled. Visitors will see increasingly detailed images as the file is downloaded. 
```

</Aside>

### How to turn on Polish

> By default, all new projects will have Polish enabled with the Lossy and WebP setting. Navigate to the Settings, then General to update or add your configuration. 

Reviewer 2: Is this happening on the CF dashboard? or somewhere else?

Author: Yea, this happens oh the CF dashboard

Reviewer 3: @deadlypants1973 it happens within the Pages dash

---
order: 10
pcx-content-type: concept
---

# Performance Improvements

> Cloudflare Pages now includes performance improvements by removing the overhead on media assets and offering image compression. 

Reviewer 2: ```suggestion
Cloudflare Pages supports performance improvement by removing the overhead on media assets and offering image compression. 
```

> Cloudflare Pages now includes performance improvements to serve pages of your projects by removing the overhead on media assets and offering compressions. 

Reviewer 2: ```suggestion
Cloudflare Pages now includes performance improvements by removing the overhead on media assets and offering image compression. 
```

 
<Aside type="note">

> If you have a custom domain set up in your Pages project, configuring your performance settings in **Speed** on the Cloudflare dashboard will override the performance settings configured in Pages. 

Reviewer 2: ```suggestion
If you have a custom domain set up in your Pages project, configuring your performance settings in **Speed** on the Cloudflare dashboard will override the performance settings configured in your **Pages project**. 
```

Reviewer 4: In the UI we are using "precedence" instead of "override" to make it clear that it doesn't impact the Pages setting itself.

What about

"performance settings configured in **Speed** on the Cloudflare dashboard will take precedence for that domain."

> If you have a custom domain tied to your Pages project, configuring your performance settings in **Speed** on the Cloudflare dashbaord will override the performance settings configured in Pages. 

Reviewer 2: ```suggestion
If you have a custom domain set up in your Pages project, configuring your performance settings in **Speed** on the Cloudflare dashboard will override the performance settings configured in Pages. 
```

</Aside>

## Brotli compression

> Cloudflare applies [Brotli](https://www.brotli.pro/) compression to help speed up page load times for your visitors. Brotli compression is selected as the preferred content-encoding method if multiple compression methods are supported by the client. If the client does not indicate that Brotli is supported, the gzip compression will be applied. 

Reviewer 2: ```suggestion
Cloudflare applies [Brotli](https://www.brotli.pro/) compression to speed up load times for your visitors. If multiple compression methods are supported by the client, Brotli compression is selected as the preferred content-encoding method. If the client does not indicate that Brotli is supported, the gzip compression will be applied. 
```

The Brotli compression option is automatically turned on for all Pages sites and cannot be turned off in the **Pages** dashboard.

> However, if your project is tied to a custom domain, you may adjust your Brotli settings in **Speed** > **Optimization** > moving the **Brotli** toggle to **Off*.

Reviewer 2: ```suggestion
However, if your Pages project has a custom domain set up, you may adjust your Brotli settings in the Cloudflare dashboard by going to **Speed** > **Optimization** > moving the **Brotli** toggle to **Off*.
```

 
You can learn more about [how Cloudflare handles compression by reviewing this support article](https://support.cloudflare.com/hc/en-us/articles/200168396-What-will-Cloudflare-compress-).

## Polish 

> Polish improves image loading time by optimizing images hosted on your `*.pages.dev`site. It reduces image file size by removing metadata (date and time, camera manufacturer, model, etc.) by compressing images where possible. Smaller images result in faster load time for images and web applications. 

Reviewer 2: ```suggestion
Polish improves image load time by optimizing images hosted on your `*.pages.dev`site. Polish reduces image file size by removing metadata (such as date, time, camera manufacturer, model) by compressing images where possible. Smaller file size results in faster load time for images and web applications. 
```

Reviewer 4: Polish will be turned on for all domains on the Pages project including pages.dev site.

How do you feel about

"Polish improves image load time by optimizing images hosted on your project's domains."

> There are two options for configuration with different performance effects on images -- _Lossless_ and _Lossy_. You can choose between these two options by going to **Speed** > **Optimization** > **Polish**.

Reviewer 2: ```suggestion
There are two options for Polish configuration -- _Lossless_ and _Lossy_ -- with different performance effects on images. You can choose between these two options by going to **Speed** > **Optimization** > **Polish**.
```

### Lossless 

> Lossless image option reduces the size of image files without impacting the visual quality. In this option, metadata is removed from PNG, GIF and JPEG but the lossless compression is done on GIF and PNG.

Reviewer 2: ```suggestion
The _Lossless_ Polish option reduces the size of image files without impacting the visual quality. With _Lossless_, metadata is removed from PNG, GIF, and JPEGs but the Lossless compression is applied to  GIF and PNGs only.
```

Reviewer 2: Is this saying that metadata is removed but compression is not applied to JPEGs? I edited the sentence to reflect this.

### Lossy 

> This Polish option reduces the file size of JPEG images using lossy compression which may reduce image quality.

Reviewer 3: ```suggestion
This Polish option reduces the file size of JPEG images using lossy compression which may reduce image quality. Using Lossy also enables WebP for your images. If the client requesting the image supports the WebP image codec, Cloudflare will serve a WebP version of the image if offering a performance advantage over the existing format.
```

moving webp info to lossy

> This Polish option reduces the file size of JPEG images using lossy compression which may reduce image quality.

Reviewer 2: ```suggestion
The _Lossy_ Polish option reduces the file size of JPEG images using Lossy compression. This option may reduce image quality.
```

<Aside heading="Note">
 

> Making the decision on what form of optimization to use is mostly based on your individual use case. You can review the variety of resources to learn more about [the difference between Lossy and Lossless image compression on Imagify's blog](https://imagify.io/blog/lossless-vs-lossy-image-compression/).

Reviewer 2: ```suggestion
Your individual use case will determine which form of optimization is right for your project. You can refer to [the difference between Lossy and Lossless image compression on Imagify's blog](https://imagify.io/blog/lossless-vs-lossy-image-compression/) to learn more about which option is right for you.
```

Reviewer 4: How do we feel about linking to Cloudflare docs instead? (https://support.cloudflare.com/hc/en-us/articles/360000607372-Using-Cloudflare-Polish-to-compress-images) In case there are differences between Cloudflare Lossless/Lossy and Imagify docs.

</Aside>

> ### Serving WebP images 
> 
> Whether you choose _Lossy_ or _Lossless_, you also have the option of enabling WebP for your images. If the client requesting the image supports the WebP image codec, Cloudflare will serve a WebP version of the image if offering a performance advantage over the existing format. 

Reviewer 3: ```suggestion
```

> ### Serving WebP images 
> 
> Whether you choose _Lossy_ or _Lossless_, you also have the option of enabling WebP for your images. If the client requesting the image supports the WebP image codec, Cloudflare will serve a WebP version of the image if offering a performance advantage over the existing format. 
> 

Reviewer 3: ```suggestion
```
removing this section to add webp description under lossy as that will be the only option supporting webp

> <Aside heading="Status Code">
> 
> Large JPEG images are converted to progressive images with WebP enabled. Visitors will see increasingly detailed images as the file is downloaded. 
> 
> </Aside>

Reviewer 3: ```suggestion
```

move this to the lossy section

### How to turn on Polish

> By default, all new projects will have Polish enabled with the Lossy and WebP setting.  Navigate to the Settings, then General to update or add your configuration. 

Reviewer 2: ```suggestion
By default, all new projects will have Polish enabled with the _Lossy_ and the **WebP** setting.  To update your configuration, log into the Cloudflare dashboard and go to **Speed** > **Optimization** > **Polish**.
```
